src/app/checkout/step-2/step-2.component.spec.js

```diff
@@ -308,6 +308,53 @@ describe('checkout', () => {
         expect(self.controller.paymentFormState).toBe('success')
         expect(self.controller.getContinueDisabled()).toBe(false)
       })
+
+      it('should return true when cvv is invalid and card type is credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = false
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(true)
+      })
+
+      it('should return false when cvv is invalid and card type is not credit card', () => {
+        self.controller.handleExistingPaymentLoading(true, true)
+        self.controller.isCvvValid = false
+        self.controller.handlePaymentChange({'account-type': 'checking'})
+        
+        expect(self.controller.getContinueDisabled()).toBe(false)
+      })
+
+      it('should return false when cvv is invalid and new payment type is credit card', () => {
+        jest.spyOn(self.controller.brandedAnalyticsFactory, 'savePaymentType')
+        self.controller.handlePaymentChange({'card-type': 'visa'})
+        self.controller.isCvvValid = false
+        expect(self.controller.getContinueDisabled()).toBe(true)
+      })
+
+      it('should return false when cvv is invalid and new payment type is bankAccount', () => {
+        jest.spyOn(self.controller.brandedAnalyticsFactory, 'savePaymentType')
+        self.controller.handlePaymentChange({'account-type': 'checking'})
+        self.controller.isCvvValid = false
```

Comment:
Will this be false or undefined? Or null?

Reply:
Gotcha, typically it is best to have a test for each possible path of the conditional, so you might want to add one more for that at least.